From dcd51362e0bf6eb22902572f8b5d85cfa286a9c8 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:55:56 -0400
Subject: [scsi] qla2xxx: Correctly set mailboxes for extended init control block

Message-id: <1408391817-6389-9-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87908
O-Subject: [PATCH RHEL 7.1 e-stor 08/69] qla2xxx: Correctly set mailboxes for extended init control block.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Joe Carnuccio <joe.carnuccio@qlogic.com>

scsi commit id 4ef21bd4857fc08edb2e53faae2adce409ca3bba

Signed-off-by: Joe Carnuccio <joe.carnuccio@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_mbx.c b/drivers/scsi/qla2xxx/qla_mbx.c
index 0708c2e..5a17ab1 100644
--- a/drivers/scsi/qla2xxx/qla_mbx.c
+++ b/drivers/scsi/qla2xxx/qla_mbx.c
@@ -1214,7 +1214,7 @@ qla2x00_init_firmware(scsi_qla_host_t *vha, uint16_t size)
  mcp->mb[6] = MSW(MSD(ha->init_cb_dma));
  mcp->mb[7] = LSW(MSD(ha->init_cb_dma));
  mcp->out_mb = MBX_7|MBX_6|MBX_3|MBX_2|MBX_1|MBX_0;
- if ((IS_QLA81XX(ha) || IS_QLA83XX(ha)) && ha->ex_init_cb->ex_version) {
+ if (ha->ex_init_cb && ha->ex_init_cb->ex_version) {
   mcp->mb[1] = BIT_0;
   mcp->mb[10] = MSW(ha->ex_init_cb_dma);
   mcp->mb[11] = LSW(ha->ex_init_cb_dma);
-- 
1.7.1